### PR TITLE
i#3544 RV64: Enable more tests on CI

### PR DESCRIPTION
**Pull request by ksco**
Enabling passed tests on RISC-V. Some extra work is necessary:

- Implements insert_parameter_preparation() and mangle_syscall_arch() by shared them with AARCHXX.
- Add currently unused SCRATCH_REG5 -- this allows us to share the code with AARCHXX more easily.
- Implements mangle_insert_clone_code()

Issue: #3544

### CODE REVIEW

**Comment by ksco** [comment]
x86-32-ubuntu22 tool.drcacheoff.simple #6416
x86-64-ubuntu22 api.rseq #6185

File: core/arch/mangle_shared.c [hunk 1]
```diff
@@ -442,6 +442,294 @@ parameters_stack_padded(void)
...
+                XINST_CREATE_store(
+                    dcontext,
+                    opnd_create_base_disp(DR_REG_XSP, DR_REG_NULL, 0, off, OPSZ_PTR),
+                    opnd_create_reg(link_reg)));
+        }
+    }
+
+    return (uint)stack_inc;
+}
+#endif /* X86 */
```
Comment (derekbruening) [inline core/arch/mangle_shared.c:731]: Just eyeballed most of this: hard to tell as a human what changed when the code was both moved to another file and slightly tweaked...

**Review by derekbruening — APPROVED** [review]

File: core/arch/mangle_shared.c (hunk 1 above)
Comment (ksco) [reply core/arch/mangle_shared.c:731]: Ah yes! I should've separated the move and tweaks into two commits, that should be easier for the review.

**Comment by ksco** [comment]
@derekbruening  vs2019-64 is failing because: [There is not enough space on the disk.](https://github.com/DynamoRIO/dynamorio/actions/runs/6859383617/job/18651613201?pr=6447#step:6:20186)

**Comment by ksco** [comment]
Oh looks like vs2019-64 is running on a standard GA runner ([Windows Server 2019](https://github.com/actions/runner-images/blob/win19/20231029.1/images/win/Windows2019-Readme.md#windows-server-2019)), not a self-hosted one, so this could be an issue on Github side.

**Comment by ksco** [comment]
x86-32-ubuntu22 tool.drcacheoff.simple failed 5 times in a row: https://github.com/DynamoRIO/dynamorio/issues/6416

**Pull request merged**